enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -357,6 +355,32 @@ func UploadTree(ctx context.Context, env environment.Env, dirHelper *DirHelper,
 				if err != nil {
 					return nil, err
 				}
+
+				ospb := &repb.OutputSymlink{
+					Path:   trimPathPrefix(fqfn, rootDir),
+					Target: target,
+				}
+				// In case the target is a dangling symlink (the target does not exist, or inaccessible),
+				// there is no way to tell whether it's a directory or a file symlink.
+				// Dangling symlinks could be used in packaging with targets that only exist on the
+				// deployment environments.
+				//
+				// Set both OutputFileSymlinks and OutputDirectorySymlinks only when the symlinks
```

Comment:
Support for unresolved symlink was added to Bazel in the recent version (since 6.0.0) https://github.com/bazelbuild/bazel/commit/8056eb566371798acdec97aedde0a2c242957ef7
https://github.com/bazelbuild/bazel/issues/10298

When a symlink is unresolved, we cannot tell whether the target is a file or a directory to add it in the respective file/dir ActionResult field.

Reply:
The current implementation seems fine (as it gives us support for resolved symlinks which we didn't have before), but I am curious how unresolved symlinks can work it we don't set them in the ActionResult.